validphys2/src/validphys/pineparser.py

```diff
@@ -155,26 +153,18 @@ def pineappl_reader(fkspec):
     # Extract metadata from the first grid
     pine_rep = pines[0]
 
-    # Is it hadronic? (at the moment only hadronic and DIS are considered)
-    try:
-        parton1 = pine_rep.key_values()["convolution_particle_1"]
-        parton2 = pine_rep.key_values()["convolution_particle_2"]
-    except KeyError:
-        # Old pineappl FKTables used  `initial_state` instead of `convolution_particle`
-        parton1 = pine_rep.key_values()["initial_state_1"]
-        parton2 = pine_rep.key_values()["initial_state_2"]
-    hadronic = parton1 == parton2
-
-    # NOTE: while the following can accept any number of convolutions, at the moment only
-    # 1 (DIS) or 2 (hadronic) are implemented.
-    # In the case of DIS grids, convolution 1 refers to the hadron
-    conv_types = [pine_rep.key_values().get("convolution_type_1", "UnpolPDF")]
-    if hadronic:
-        # Sanity check (in case at some point we start fitting things that are not protons)
-        if parton1 != "2212":
-            raise ValueError("vp can only read hadronic fktables with 2 protons!")
+    # Get the convolution types for this FK table
+    convolutions = pine_rep.convolutions
+
+    # Is it hadronic? For the time being, hadronic is defined with `len(convolutions) == 2`.
+    # Hadronic could also involve 3 convolutions in processes such as `pp->H` (with FFs).
+    # DIS FK table now only contains ONE single convolutions.
+    hadronic = len(convolutions) == 2
```

Comment:
The pid type is important to check to avoid problems like unknowingly fittingna p pbar fktable for instance

Reply:
Good point! I will modify this accordingly.